libvast/include/vast/detail/inspection_common.hpp

```diff
@@ -0,0 +1,88 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include <string_view>
+#include <utility>
+
+namespace vast::detail {
+
+template <class Inspector, class AfterInspectionCallback>
+class inspection_object_with_after_inspection_callback {
```

Comment:
The types defined in this file need to be documented. As it is I don't understand when I'm supposed to use them.